core/src/actions/setupSteps.ts

```diff
@@ -12,6 +12,11 @@ export class SetupStepsList extends AuthenticatedAction {
     this.outputExample = {};
   }
 
+  isWriteTransaction() {
+    // setupStep.performCheck() can do an update
+    return true;
```

Comment:
should the `permission` be `write` - I wasn't sure.

Reply:
Yeah - `await setupSteps[i].performCheck()` will potentially mark a setup step as complete in the DB.  Alternatively, checking steps could be moved to a task maybe?  Doing it here made the UI more responsive to changes...